playbooks/ansible-tox-py39/pre.yaml

```
@@ -1,8 +1,22 @@
 ---
 - hosts: all
   tasks:
-    - name: Ensure python3.9 is present
+
+    - name: Load operating system specific variables
+      include_vars: "{{ item }}"
+      failed_when: false
+      loop:
+          - "family-{{ ansible_os_family | lower }}.yml"
+          - "family-{{ ansible_os_family | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
+          - "{{ ansible_distribution | lower | replace(' ', '-') }}.yml"
+          - "{{ ansible_distribution | lower | replace(' ', '-') }}-{{ ansible_distribution_major_version | lower }}.yml"
+          - "{{ ansible_distribution | lower | replace(' ', '-') }}-{{ ansible_distribution_version.split('.')[0:2] | join('-') | lower }}.yml"
+          - "{{ ansible_distribution | lower | replace(' ', '-') }}-{{ ansible_distribution_version.split('.')[0:3] | join('-') | lower }}.yml"
```

As written, this will include _all_ of these files and error if one does not exists (which is why ignoring errors is necessary as implemented).

Usually the strategy with including distribution specific variables is to include the first found file, from most specific to least specific. You can provide an empty file for `default.yml` or allow it to fail if no files are found.

```suggestion
      include_vars: "{{ lookup('first_found', params) }}"
      vars:
        params:
          files:
            - "{{ ansible_facts.distribution }}-{{ ansible_facts.distribution_version }}.yml"
            - "{{ ansible_facts.os_family }}-{{ ansible_facts.distribution_major_version }}.yml"
            - "{{ ansible_facts.distribution }}.yml"
            - "{{ ansible_facts.os_family }}.yml"
            - default.yml
          paths:
            - vars
```

---

Or instead of debating over subjective nits like lowercase, or order of loading maybe it would just be better to revert the breaking change from https://github.com/ansible/ansible-zuul-jobs/pull/1201?

Clearly that went in without serious testing as it would clearly break on both fedora 34 and 35.

---

FYI my approach proves to minimize maintenance for complex setups where you want to only override settings by more specific selectors. With your approach if a new variable needs to be added for all redhat products, you may have to add it to 10 files if that is the number of variances across redhat platforms.

With layered approach, you can minimize the number of places where you define a variable.